context.go

```diff
@@ -1399,6 +1423,57 @@ func (c *Context) addInterVariantDependency(origModule *moduleInfo, tag Dependen
 	atomic.AddUint32(&c.depsModified, 1)
 }
 
+// parentFile returns the path of a file with the same name but in a parent directory
+// For example, if path = "a/b/Blueprints", then parent = "a/Blueprints"
+func parentFile(path string) (parent string, hasParent bool) {
```

Comment:
Would it be easier to do the ancestor mapping on directories and ignore the file name?

Reply:
done